src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -1,263 +1,211 @@
+import os
 import os.path
+import shutil
+from collections import defaultdict
+from dataclasses import dataclass, field, replace
 from typing import Dict, List, Optional
 
-from cwrap import BaseCClass
-from ecl.util.util import StringList
-
-from ert._c_wrappers import ResPrototype
-from ert._c_wrappers.config import ContentTypeEnum
+from ert._c_wrappers.config import ConfigParser, ConfigValidationError, ContentTypeEnum
 from ert._c_wrappers.util import SubstitutionList
+from ert._clib.job_kw import type_from_kw
+
+
+@dataclass
```

Comment:
I am not really sure these types of objects should be forced to be dataclasses as far as I can tell a dataclass should be only a data object that has very little logic (hopefully none) around the data fields as it is implemented now it does not really fit that definition.

Reply:
ext_job fits that description perfectly though. There is just a static method that creates the object from a file.

Reply:
Also, we talked about eventually turning ResConfig into a pydantic model/dataclass. Then making ext_job into an equivalent of a dataclass is necessary.